aiida/backends/tests/orm/data/test_singlefile.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for the `SinglefileData` class."""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+import tempfile
+
+from aiida.backends.testbase import AiidaTestCase
+from aiida.orm import SinglefileData, load_node
+
+
+class TestSinglefileData(AiidaTestCase):
+    """Tests for the `SinglefileData` class."""
+
+    def test_reload_singlefile_data(self):
+        """Test writing and reloading a `SinglefileData` instance."""
+        content_original = 'some text ABCDE'
+
+        with tempfile.NamedTemporaryFile(mode='w+') as handle:
+            filepath = handle.name
+            basename = os.path.split(filepath)[1]
```

Comment:
why not use os.path.basename as below?

Reply:
That test already existed and merely copied it from `test_basedataclasses.py` and simply didn't adapt it. I will change it